src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -147,7 +147,13 @@ public void install_noop(ChocolateyConfiguration config)
             // each package can specify its own configuration values    
             foreach (var packageConfig in set_config_from_package_names_and_packages_config(config, new ConcurrentDictionary<string, PackageResult>()).or_empty_list_if_null())
             {
-                _nugetService.install_noop(packageConfig, (pkg) => _powershellService.install_noop(pkg));
+                _nugetService.install_noop(packageConfig, (pkg) =>
+                    {
+                        foreach (var installProvider in _installProviders.Where(ip => ip.can_be_used(config)).or_empty_list_if_null())
+                        {
+                            installProvider.install_noop(pkg);

```

Comment:
We need a way to shortcut out once one runs so that not more than one would run.